pkg/component/logging/fluentoperator/fluent_bit.go

```diff
@@ -0,0 +1,180 @@
+				Name:      v1beta1constants.DaemonSetNameFluentBit + "-lua-config",
+				Namespace: f.namespace,
+			},
+			Data: map[string]string{
+				"modify_severity.lua": `
+function cb_modify(tag, timestamp, record)
+  local unified_severity = cb_modify_unify_severity(record)
+
+  if not unified_severity then
+    return 0, 0, 0
+  end
+
+  return 1, timestamp, record
+end
+
+function cb_modify_unify_severity(record)
+  local modified = false
+  local severity = record["severity"]
+  if severity == nil or severity == "" then
+	return modified
+  end
+
+  severity = trim(severity):upper()
+
+  if severity == "I" or severity == "INF" or severity == "INFO" then
+    record["severity"] = "INFO"
+    modified = true
+  elseif severity == "W" or severity == "WRN" or severity == "WARN" or severity == "WARNING" then
+    record["severity"] = "WARN"
+    modified = true
+  elseif severity == "E" or severity == "ERR" or severity == "ERROR" or severity == "EROR" then
+    record["severity"] = "ERR"
+    modified = true
+  elseif severity == "D" or severity == "DBG" or severity == "DEBUG" then
+    record["severity"] = "DBG"
+    modified = true
+  elseif severity == "N" or severity == "NOTICE" then
+    record["severity"] = "NOTICE"
+    modified = true
+  elseif severity == "F" or severity == "FATAL" then
+    record["severity"] = "FATAL"
+    modified = true
+  end
+
+  return modified
+end
+
+function trim(s)
+  return (s:gsub("^%s*(.-)%s*$", "%1"))
+end`,
+				"add_tag_to_record.lua": `
+function add_tag_to_record(tag, timestamp, record)
+  record["tag"] = tag
+  return 1, timestamp, record
+end`,
+			},
+		}
+	)
+
+	utilruntime.Must(kubernetesutils.MakeUnique(configMap))
+
+	resources := []client.Object{configMap}
+
+	fluentBit := customresources.GetFluentBit(getFluentBitLabels(), v1beta1constants.DaemonSetNameFluentBit, f.namespace, f.values.Image, f.values.InitContainerImage, f.values.PriorityClass)
+	resources = append(resources, fluentBit)
+
+	clusterFluentBitConfig := customresources.GetClusterFluentBitConfig(v1beta1constants.DaemonSetNameFluentBit, getCustomResourcesLabels())
+	resources = append(resources, clusterFluentBitConfig)
+
+	for _, clusterInput := range customresources.GetClusterInputs(getCustomResourcesLabels()) {
+		resources = append(resources, clusterInput)
+	}
+
+	for _, clusterFilter := range customresources.GetClusterFilters(configMap.Name, getCustomResourcesLabels()) {
+		resources = append(resources, clusterFilter)
+	}
+
+	for _, clusterParser := range customresources.GetClusterParsers(getCustomResourcesLabels()) {
+		resources = append(resources, clusterParser)
+	}
+
+	resources = append(resources, customresources.GetDefaultClusterOutput(getCustomResourcesLabels()))
+
+	serializedResources, err := registry.AddAllAndSerialize(resources...)
+	if err != nil {
+		return err
+	}
+
+	return managedresources.CreateForSeed(ctx, f.client, f.namespace, FluentBitManagedResourceName, false, serializedResources)
```

Comment:
Switching to new `ManagedResource`s in general requires more actions, e.g.:
1. Change class of existing `ManagedResource`s
2. Patch out finalizer and delete "old" `ManagedResource`s
3. Create new `ManagedResource`s